docs/modules/ROOT/pages/tmail-backend/jmap-extensions/pushWithFirebase.adoc

```diff
@@ -0,0 +1,210 @@
+= Push with Firebase
+:navtitle: Push with Firebase
+
+Allows devices to register PUSH though link:https://firebase.google.com/[FireBase Cloud Messaging], and manage its
+registrations. The server will then push `StateChanges` through FireBase to the client.
+
+== Additions to the capability object
+
+Servers supporting the Push with firebase extension need
+to advertize it through the session capabilities:
+
+....
+com:linagora:params:jmap:firebase:push
+....
+
+The associated object is empty.
+
+== Encoding of state change into firebase payload
+
+Once registered, the JMAP server will send StateChange objects to firebase using the token of the Firebase registration.
+
+See https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#Message.FIELDS.data for Firebase message format.
+
+StateChange is encoded into the *data* part of the payload.
```

Comment:
there are some kinds of messages

<img width="965" alt="image" src="https://user-images.githubusercontent.com/6462404/194036137-e314e042-4c7e-4cc2-bf14-70e74e5ed9fa.png">

Please make sure back-end will send `data message`, because client only handle that message before system show notification

Reply:
Using this we loose indempotence. IE dropping messages becomes a real problem.

Using this, RGDP wise we have private data transiting through GAFAM wich is a no-no.

Using this we no longer stick to the JMAP protocol. 

Let's stick to data only nheeee